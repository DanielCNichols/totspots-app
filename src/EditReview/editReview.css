.edit_form_component h3 {
  text-align: center;
  margin: 15px, auto;
}

.edit_section {
  margin: 100px auto;
}

.edit_form {
  width: 100%;
  margin-left: auto;
  margin-right: auto;
}
.edit_form label {
  text-align: center;
  display: block;
  margin: 1.5em;
}

.edit_form_text {
  display: block;
  width: 80%;
  margin: 5px auto;
  padding: 5px 10px;
  height: 2.5rem;
  min-width: 125px;
  border: 1px solid black;
  border-left: 3px solid;
  border-radius: 5px; 
  transition: border-color .5s ease-out;
}

.edit_form_select {
  display: block;
  width: 90%;
  padding: 5px 10px;
  height: 2.5rem;
  border-radius: 5px;
  margin: 5px auto;
  cursor: pointer;
}

legend {
  text-align: center;
  width: 100%;
  margin: 0 auto;
}

.amenities {
  display: flex;
  flex-direction: column;
}

.amenities label {
  display: flex;
  justify-content: space-between;
  margin: 10px, auto;
}

.amenities label > input {
  margin-left: 20px;
  cursor: pointer;
}

fieldset {
  border: none;
}

.review {
  width: 80%;  
  margin: 0 auto;
}

.review label {
  text-align: left;
  margin: 1.5em;
}

.review textarea { 
  display: block;
  margin: 5px auto;
  padding: 5px 10px;
  width: 50%;
  height: 5rem;
  min-width: 125px;
  border: 1px solid black;
  border-left: 3px solid;
  border-radius: 5px; 
  transition: border-color .5s ease-out;
}


.edit_form_controls {
  width: 80%;
  margin: 0 auto;
}

.edit_form_controls button {
  display: block;
  min-width: 180px;
  height: 40px;
  margin: 20px auto;
  border-radius: 5px;
  cursor: pointer;
  background-color: transparent;
  font-size: 1.1em;
}

.edit_form_button {
  border-color: #064789;
}

.edit_form_cancel {
  border-color: #ED254E;
}

@media screen and (min-width: 450px) {
  .edit_form {
    width: 75%;
    margin: auto;
  }

  .edit_form_controls {
    display: flex;
    justify-content: center;
  }

  .edit_form_controls button {
    display: inline-block;
    width: 20px;
  }
}

@media screen and (min-width: 800px) {
  .edit_section {
    width: 60%;
  }


  .edit_form {
    width: 50%;
  }

  .edit_form_text {
    width: 80%;
  }

  .edit_form_label {
    text-align: left;
  }
}