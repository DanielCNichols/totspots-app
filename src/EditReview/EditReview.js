import React, { Component } from 'react';
import PropTypes from 'prop-types';
import VenuesContext from '../VenuesContext';
import config from '../config';
import ApiService from '../services/api-service';
import './editReview.css'

const Required = () => <span className='EditBookmark__required'>*</span>;

class editReview extends Component {
  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.object
    }),
    history: PropTypes.shape({
      push: PropTypes.func
    }).isRequired
  };

  static contextType = VenuesContext;

  state = {
    error: null,
    id: '',
    venue_id: '',
    content: '',
    price: '',
    volume: '',
    starrating: 1,
    user_id: ''
  };

  componentDidMount() {
    const reviewId = this.props.match.params.review_id;
    ApiService.getReview(reviewId).then(responseData => {
      this.setState({
        id: responseData.id,
        venue_id: responseData.venue_id,
        content: responseData.content,
        price: responseData.price,
        volume: responseData.volume,
        starrating: responseData.starrating
      });
    });
  }

  handleChangeContent = e => {
    this.setState({ content: e.target.value });
  };

  handleChangePrice = e => {
    this.setState({ price: e.target.value });
  };

  handleChangeVolume = e => {
    this.setState({ volume: e.target.value });
  };

  handleChangeRating = e => {
    this.setState({ starrating: e.target.value });
  };

  handleSubmit = e => {
    e.preventDefault();
    // const { reviewId } = this.props.match.params;
    const { id, venue_id, content, price, volume, starrating } = this.state;
    const newReview = { id, venue_id, content, price, volume, starrating };
    ApiService.editReview(id, newReview)
      .then(() => {
        this.resetFields(newReview);
        this.context.updateReview(newReview);
        this.props.history.goBack();
      })
      .catch(error => {
        console.error(error);
        this.setState({ error });
      });
  };

  resetFields = newReview => {
    this.setState({
      id: newReview.id || '',
      venue_id: newReview.title || '',
      content: newReview.url || '',
      price: newReview.description || '',
      volume: newReview.rating || '',
      starrating: newReview.starrating || ''
    });
  };

  handleClickCancel = () => {
    this.props.history.push('/');
  };

  render() {
    const {
      error,
      content,
      price,
      volume,
      starrating
    } = this.state;
    return (
      <section>
        <header>
          <h3>Edit Review</h3>
        </header>
        <form className="edit_form" onSubmit={this.handleSubmit}>
          <fieldset>
              <label className="edit_form_label" htmlFor='price'>
                Price
                <select
                value= {price}
                className="edit_form_select"
                  name='price'
                  id='price'
                  onChange={this.handleChangePrice}
                >
                  <option value=''>Please select</option>
                  <option value='1'>$</option>
                  <option value='2'>$$</option>
                  <option value='3'>$$$</option>
                  <option value='4'>$$$$</option>
                  <option value='5'>$$$$$</option>
                </select>
              </label>
              <label 
              className="edit_form_label"
              htmlFor='volume'>
                Volume Level
                <select
                value = {volume}
                 className="edit_form_select"
                  name='volume'
                  id='volume'
                  onChange={this.handleChangeVolume}
                >
                  <option value=''>Please select</option>
                  <option value='1'>Library</option>
                  <option value='2'>Coffee Shop</option>
                  <option value='3'>Restaurant</option>
                  <option value='4'>Bar/Brewery</option>
                  <option value='5'>Concert</option>
                </select>
              </label>
              <label 
              className="edit_form_label"
              htmlFor='rating'>
                Overall rating
                <select
                value = {starrating}
                  className="edit_form_select"
                  name='rating'
                  id='rating'
                  onChange={this.handleChangeRating}
                >
                  <option value=''>Please select</option>
                  <option value='1'>&#x2605;</option>
                  <option value='2'>&#x2605; &#x2605;</option>
                  <option value='3'>&#x2605; &#x2605; &#x2605;</option>
                  <option value='4'>&#x2605; &#x2605; &#x2605; &#x2605;</option>
                  <option value='5'>
                    &#x2605; &#x2605; &#x2605; &#x2605; &#x2605;
                  </option>
                </select>
              </label>
          </fieldset>
          <div className='review'>
            <label 
            className="edit_form_label"
            htmlFor='review'>
              What's changed at this spot? 
              <textarea
              className="edit_form_text"
                type='text'
                name='content'
                value={content}
                placeholder='Tell us about your visit'
                onChange={this.handleChangeContent}
              />
            </label>
          </div>
          <div className="edit_form_controls">
            <button className="edit_form_button" type='submit'>Submit</button>
            <button className="edit_form_cancel" onClick={this.handleClickCancel}>Cancel</button>
          </div>
        </form>
      </section>
    );
  }
}
export default editReview;
