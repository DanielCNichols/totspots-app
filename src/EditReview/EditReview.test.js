import React from 'react';
import ReactDOM from 'react-dom'
import EditReview from './EditReview'
import renderer from 'react-test-renderer'
import {MemoryRouter} from 'react-router-dom'




it('Edit Review renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <EditReview match={{params: {review_id:25}}}/>
    </MemoryRouter>
    , div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders Edit Review as expected', () => {
  const tree = renderer
  .create(
    <MemoryRouter>
      <EditReview  match={{params: {review_id:25}}}/>
    </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
