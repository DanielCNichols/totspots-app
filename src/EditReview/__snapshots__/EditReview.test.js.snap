// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Edit Review as expected 1`] = `
<section
  className="edit_section"
>
  <header>
    <h3>
      Edit Review
    </h3>
  </header>
  <div
    role="alert"
  />
  <form
    className="edit_form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        <label
          className="edit_form_label"
          htmlFor="price"
        >
          Price
          <select
            className="edit_form_select"
            id="price"
            name="price"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Please select
            </option>
            <option
              value="1"
            >
              $
            </option>
            <option
              value="2"
            >
              $$
            </option>
            <option
              value="3"
            >
              $$$
            </option>
            <option
              value="4"
            >
              $$$$
            </option>
            <option
              value="5"
            >
              $$$$$
            </option>
          </select>
        </label>
        <label
          className="edit_form_label"
          htmlFor="volume"
        >
          Describe the volume level
          <select
            className="edit_form_select"
            id="volume"
            name="volume"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Please select
            </option>
            <option
              value="1"
            >
              Library
            </option>
            <option
              value="2"
            >
              Coffee Shop
            </option>
            <option
              value="3"
            >
              Restaurant
            </option>
            <option
              value="4"
            >
              Bar/Brewery
            </option>
            <option
              value="5"
            >
              Concert
            </option>
          </select>
        </label>
        <label
          className="edit_form_label"
          htmlFor="rating"
        >
          Overall rating
          <select
            className="edit_form_select"
            id="rating"
            name="rating"
            onChange={[Function]}
            value={1}
          >
            <option
              value=""
            >
              Please select
            </option>
            <option
              value="1"
            >
              ★
            </option>
            <option
              value="2"
            >
              ★ ★
            </option>
            <option
              value="3"
            >
              ★ ★ ★
            </option>
            <option
              value="4"
            >
              ★ ★ ★ ★
            </option>
            <option
              value="5"
            >
              ★ ★ ★ ★ ★
            </option>
          </select>
        </label>
      </legend>
    </fieldset>
    <div
      className="edit_review"
    >
      <label
        className="edit_form_label"
        htmlFor="review"
      >
        What's changed at this spot?
        <textarea
          className="edit_form_text"
          name="content"
          onChange={[Function]}
          placeholder="Tell us about your visit"
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="edit_form_controls"
    >
      <button
        className="edit_form_cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="edit_form_button"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
